ansible_base/authentication/serializers/authenticator.py

```diff
@@ -1,27 +1,46 @@
 from collections import OrderedDict
 
+from django.utils.text import slugify
 from django.utils.translation import gettext_lazy as _
 from rest_framework.serializers import ChoiceField, ValidationError
 
-from ansible_base.authentication.authenticator_plugins.utils import get_authenticator_plugin, get_authenticator_plugins
+from ansible_base.authentication.authenticator_plugins.utils import generate_authenticator_slug, get_authenticator_plugin, get_authenticator_plugins
 from ansible_base.authentication.models import Authenticator
 from ansible_base.lib.serializers.common import NamedCommonModelSerializer
+from ansible_base.lib.serializers.mixins import ImmutableFieldsMixin
 from ansible_base.lib.utils.encryption import ENCRYPTED_STRING
 from ansible_base.lib.utils.response import get_relative_url
 
 
-class AuthenticatorSerializer(NamedCommonModelSerializer):
+class AuthenticatorSerializer(NamedCommonModelSerializer, ImmutableFieldsMixin):
     type = ChoiceField(get_authenticator_plugins())
 
     def validate_type(self, value):
         if self.instance and self.instance.type != value:
             raise ValidationError(_("Cannot change authenticator type after it has been created."))
         return value
 
+    def validate_slug(self, value):
+        value = slugify(value)
```

Comment:
If we alter the get_authenticator_slug we can call it here instead. This would also get rid of the else statement.